yjit/src/asm/mod.rs

```diff
@@ -142,6 +147,34 @@ impl CodeBlock
         self.write_pos + num_bytes < self.mem_size
     }
 
+    /// Add an assembly comment if the feature is on.
+    /// If not, this becomes an inline no-op.
+    #[inline]
+    pub fn add_comment(&mut self, comment: String) {
+        #[cfg(feature="asm_comments")]
+        {
+            let is_dup = if let Some((last_pos, last_comment)) = self.asm_comments.last() {
+                *last_pos == self.write_pos && *last_comment == comment
+            } else { false };
+            if !is_dup {
+                self.asm_comments.push((self.write_pos, comment));
+            }
+        }
+    }
+
+    /// Add an assembly comment at a specific byte offset if the feature is on.
+    /// If not, this becomes an inline no-op.
+    #[inline]
+    pub fn add_comment_at(&mut self, pos:usize, comment: String) {
+        #[cfg(feature="asm_comments")]
+        self.asm_comments.push((pos, comment));
+    }
+
+    /// Get all assembly comments. If the feature is off, this will be empty.
+    pub fn get_comments(&self) -> Vec<(usize, String)> {
+        self.asm_comments.clone()
+    }
```

Comment:
Hm this should probably return an iterator if that's possible, because if I understand correctly, this is copying the entire vector, which is relatively expensive.